src/foundry/client/data/collections/compendium-collection.d.mts

```diff
@@ -50,7 +51,7 @@ declare global {
     T extends CompendiumCollection.Metadata,
   > extends DirectoryCollectionMixin_DocumentCollection<Document.ConfiguredClassForName<T["type"]>, T["name"]> {
     /** @param metadata - The compendium metadata, an object provided by game.data */
-    constructor(metadata: T);
+    constructor(metadata: CompendiumCollection.ConstructorMetadata<T["type"]>);
```

Comment:
I see two options here:
- `CompendiumCollection` should be generic over `T extends foundry.CONST.COMPENDIUM_DOCUMENT_TYPES` instead of `T extends CompendiumCollection.Metadata`
- `CompendiumCollection.ConstructorMetadata` and such should stop being generic and instead be merged with `T`. Here that'd be `constructor(metadata: T & CompendiumCollection.ConstructorMetadata)`.

Option #1 is suitable if we think no one will subclass `CompendiumCollection` with extra metadata. Option #2 makes most sense to me otherwise.

Reply:
Ideally I think T would extend `CompendiumCollection.Metadata` because then if you needed to explicitly use the type, it would have the same structure as all the other documents.  The issue with merging `CompendiumCollection.Metadata` into the others, though, is that it contains a bunch of required fields that the others don't.  For #2, are you suggesting stripping `Metadata` down to fewer fields?  That could work, though then `Metadata` wouldn't look like people might expect.  I could make T extend another, smaller interface with just the common fields and then have all the others extend it.  That would also make it ugly to have to explicitly use the type, though.

I did look at making the interfaces not generic, and having T extend `CompendiumCollection.MetaData` without merging them.  I think that works other than it loses the connection of ensuring that the compendium types match across them, but maybe that's a small price to pay for simplicity?

I'll also add some tests next time around, because I should have caught this issue.

Reply:
I believe `SimpleMerge` would address this concern.

I will say that it having the same structure as all the other documents doesn't seem that valuable to me though, which might indicate it should be the first. I don't think people will tend to be expecting `Actor` and `CompendiumCollection` to have the same type parameters. They definitely won't in fact.

Maybe I'm missing which classes you're trying to make it most similar to though.